### PR TITLE
Add Temporary Station of Jurisdiction endpoint to scv-bip-api and mock-bip-claims-api

### PR DESCRIPTION
**Pull request by dfitchett**
## What was the problem?

Current svc-bip-api does not contain endpoint to set the `temp_station_of_jurisdiction`

Associated tickets or Slack threads:
- #2175 
- Blocked by / Waiting on merges of:
  - #2144 
  - #2165

## How does this fix it?[^1]

This adds the endpoint for setting the temporary station of jurisdiction to the BipApiService and the associated rabbitMQ configurations/declarations and listeners. 
This follows the class paradigm as diagramed in  #2165

## How to test this PR
- Run the unit tests via `./gradlew :svc-bip-api:check`
- Start the platform base docker containers to start RabbitMq and start the BIP Mock container (see [Docker Compose](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose))
  - run the integration tests via `./gradlew :svc-bip-api:integrationTest`
  - build and compose the `svc-bip-api` docker and run a test by:
     - create a new queue on the `bip-api` exchange called `response`
     - open to the rabbitmq console to the [getClaimDetailsQueue](http://localhost:15672/#/queues/%2F/getClaimDetailsQueue) 
     - publish a message with the body `1000` and add a message property called `reply_to` with the value `response`
     - open the response queue page and perform the get message action
     - verify it is a 200 and the response is the mocked response containing: `{"statusCode":200,"statusMessage":"OK","claim":{"claimId":"1000","phase":"Claim Received","endProductCode":"400"}}`
     - open to the rabbitmq console to the [putTempStationOfJurisdictionQueue](http://localhost:15672/#/queues/%2F/putTempStationOfJurisdictionQueue) 
     - publish a message with the body `{"claimId":1000,"tempStationOfJurisdiction":"398"` and add a message property called `reply_to` with the value `response`
     - open the response queue page and perform the get message action
     - verify it is a 200 with the response body `{"statusCode":200,"statusMessage":"OK"}`
     - open to the rabbitmq console to the [getClaimDetailsQueue](http://localhost:15672/#/queues/%2F/getClaimDetailsQueue) 
     - publish a message with the body `1000` and add a message property called `reply_to` with the value `response`
     - open the response queue page and perform the get message action
     - verify it is a 200 and the response is the mocked response containing: `{"statusCode":200,"statusMessage":"OK","claim":{"claimId":"1000","phase":"Claim Received","endProductCode":"400","tempStationOfJurisdiction":"398"}}`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py
Comment (greene858) [inline domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py:1]: Is the TemporaryStationOfJuridiction only used with EP400 claims?

Comment (dfitchett) [reply domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py:1]: Its actually a field in the claimDetails for every claim. However, here, we are modifying the tempStationOfJurisdiction only on the EP400.

File: svc-bip-api/src/main/java/gov/va/vro/bip/config/RabbitMqConfig.java [hunk 1]
```diff
@@ -46,4 +57,47 @@ RabbitListenerErrorHandler svcBipApiErrorHandler() {
         };
     return handler;
   }
+
+  @Bean
+  RabbitListenerErrorHandler svcBipApiErrorHandlerV2(ObjectMapper mapper) {
```
Comment (raudabaugh) [inline svc-bip-api/src/main/java/gov/va/vro/bip/config/RabbitMqConfig.java:62]: What's the purpose of having two separate error handler versions? When should I use one over the other?

**Comment by raudabaugh** [comment]
> This follows the class paradigm as diagramed in

I think you forgot to include a link in the PR description?

**Review by msnwatson — COMMENTED** [review]
Is there anyway to break this PR into smaller pieces 😅

File: svc-bip-api/src/integrationTest/java/gov/va/vro/bip/RabbitMqIntegrationTest.java [hunk 2]
```diff
@@ -110,4 +100,11 @@ private void assertResponseIsSuccess(BipUpdateClaimResp response) {
     Assertions.assertFalse(textFields.isEmpty());
     Assertions.assertTrue(textFields.contains("Success"));
   }
+
+  private void assertResponseIsSuccess(BipPayloadResponse response) {
```
[the remark this answers is not in the document]
Comment (dfitchett) [reply svc-bip-api/src/integrationTest/java/gov/va/vro/bip/RabbitMqIntegrationTest.java:104]: My goal is to refactor the RabbitMqIntegrationTest out of existence by splitting up the endpoints into their own tests classes like the GetClaimContentionsTest and the PutTempStationOfJurisdictionTest. We still need to refactor the other endpoints included in that test.

File: svc-bip-api/src/main/java/gov/va/vro/bip/config/RabbitMqConfig.java (hunk 1 above)
Comment (dfitchett) [reply svc-bip-api/src/main/java/gov/va/vro/bip/config/RabbitMqConfig.java:62]: The old one will be refactored out of existence when the refactors are completed for the remaining endpoints.

**Comment by dfitchett** [comment]
> Is there anyway to break this PR into smaller pieces 😅

Yeah totally. I was gonna break it into two PRs. One for the mock and the other for the svc-bip-api. @greene858 mentioned rather having it all together!

**Review by nanotone — APPROVED** [review]
The `domain-ee` changes look good to me.

**Review by chengjie8 — APPROVED** [review]
LGTM. Great work!

File: mocks/mock-bip-claims-api/src/main/java/gov/va/vro/mockbipclaims/controller/BaseController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import gov.va.vro.mockbipclaims.model.bip.ProviderResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.time.OffsetDateTime;
+
+public class BaseController {
+  protected static final int CLAIM_YIELDS_500 = 500;
+
+  protected <T extends ProviderResponse> ResponseEntity<T> create200(T response) {
```
Comment (agile-josiah) [inline mocks/mock-bip-claims-api/src/main/java/gov/va/vro/mockbipclaims/controller/BaseController.java:13]: Is there value to merging the methods into a single method with a switch case on a status parameter that's passed in from the calling method? Such as changing to a createResponse(response, httpstatus) method instead.

File: svc-bip-api/src/test/java/gov/va/vro/bip/config/BipApiConfigTest.java [hunk 4]
```diff
@@ -18,30 +18,58 @@
...
+      assertTrue(e.getCause() instanceof NullPointerException);
+    }
+  }
+
+  @Test
+  public void testGetHttpsRestTemplate_WithBadCerts() {
+    Assertions.assertThrows(
+        BipException.class,
+        () -> {
+          config.setTrustStore("biptruststore.jks");
```
Comment (agile-josiah) [inline svc-bip-api/src/test/java/gov/va/vro/bip/config/BipApiConfigTest.java:45]: Can "biptruststore.jks" and others become a global constant?

**Review by agile-josiah — CHANGES_REQUESTED** [review]

**Review by agile-josiah — APPROVED** [review]

**Comment by agile-josiah** [comment]
LGTM, great work! @dfitchett

**Pull request merged**